app/services/imo_api_service.rb

```diff
@@ -0,0 +1,71 @@
+class ImoApiService
+  IMO_USERNAME = ENV["IMO_USERNAME"]
+  IMO_PASSWORD = ENV["IMO_PASSWORD"]
+  BASE_URL = "https://sgp.imo.im/api/simple/"
+
+  class ImoApiService::HTTPError < HTTP::Error
+  end
+
+  attr_reader :phone_number, :recipient_name, :locale
+
+  def initialize(phone_number:, recipient_name:, locale:)
+    @phone_number = phone_number
+    @recipient_name = recipient_name
+    @locale = locale
+  end
+
+  def invite
+    url = BASE_URL + "send_invite"
+    request_body = JSON(
+      phone: phone_number,
+      msg: invitation_message,
+      contents: [{key: "Name", value: recipient_name}, {key: "Notes", value: invitation_message}],
+      title: "Invitation",
+      action: "Click here"
+    )
+    response = execute_post(url, body: request_body)
+    result = process_response(response)
+    if result == "failure"
+      report_error(description: "Failed Imo invitation", api_path: url, response: response)
+    end
+    result
+  end
+
+  private
+
+  def execute_post(url, data)
+    HTTP
+      .basic_auth(user: IMO_USERNAME, pass: IMO_PASSWORD)
+      .post(url, data)
+  rescue HTTP::Error => e
+    report_error(description: "Error while calling the Imo API", api_path: url, exception: e)
+    raise ImoApiService::HTTPError
+  end
+
+  def process_response(response)
+    return "invited" if response.status == 200
+    if response.status == 400
+      parsed = JSON.parse(response.body)
+      error_type = parsed.dig("response", "type")
+      return "no_imo_account" if error_type == "nonexistent_user"
+    end
+    "failure"
+  end
```

Comment:
I think we could refactor to something like this if my understanding of the API and control flow is correct:

```suggestion
  def process_response(response, path)
    case response.status
    when 200 then :invited
    when 400 then 
      if JSON.parse(response.body).dig("response", "type") == "nonexistent_user"
        :no_imo_account
      else
        raise Error.new("Unknown 400 error from IMO", path: path, response: response)
      end
    else
      raise Error.new("Unknown response error from IMO", path: path, response: response)
    end
```